core/src/main/java/org/apache/calcite/sql/validate/SqlUserDefinedAggFunction.java

```diff
@@ -54,7 +54,8 @@ public SqlUserDefinedAggFunction(SqlIdentifier opName,
       SqlOperandTypeInference operandTypeInference,
       SqlOperandTypeChecker operandTypeChecker, AggregateFunction function,
       boolean requiresOrder, boolean requiresOver, RelDataTypeFactory typeFactory) {
-    super(Util.last(opName.names), opName, SqlKind.OTHER_FUNCTION,
+    super(Util.last(opName.names), opName, SqlKind.getSqlKindByName(opName.names.get(0)) != null
+                    ? SqlKind.valueOf(opName.names.get(0)) : SqlKind.OTHER_FUNCTION,
```

Comment:
@julianhyde .I see, if this is not OK,  i can change another method. The reason that causes this problem is that in parser phase, the parser directly convert 'SUM' to SqlSumAggFunction not 
SqlUnresolvedFuntion by name and at this time user-defined aggregate function has not be register. this should be converted in the validate phase.  Converting in the early phase may accelerate the validate phase later, but it will omit the user-defined aggregate function, which will lead the problem that validator can't find function.
My resolution:
Method of createCall in  SqlAbstractParserImpl.java directly returns SqlUnresolvedFunction not a clear specific funtion.
is that ok? thanks